src/ProducerEventProcessor/ProducerEventProcessor.cpp

```diff
@@ -536,9 +593,14 @@ void ProducerEventProcessorImpl::ProcessThreadNamesSnapshotAndTransferOwnership(
 
 void ProducerEventProcessorImpl::ProcessThreadStateSliceAndTransferOwnership(
     ThreadStateSlice* thread_state_slice) {
-  ClientCaptureEvent event;
-  event.set_allocated_thread_state_slice(thread_state_slice);
-  client_capture_event_collector_->AddEvent(std::move(event));
+  if (thread_state_slice->switch_out_or_wakeup_callstack_status() ==
```

Comment:
We might want to check that we don't get a slice with `kCallstackSet`.

Reply:
Done.